legato-api/src/main/java/org/opendaylight/legato/LegatoServiceController.java

```diff
@@ -0,0 +1,384 @@
+import org.opendaylight.yangtools.yang.common.RpcResult;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Optional;
+
+public class LegatoServiceController extends UnimgrDataTreeChangeListener<Evc> {
+
+	public LegatoServiceController(DataBroker dataBroker) {
+		super(dataBroker);
+		registerListener();
+	}
+
+	private static final Logger LOG = LoggerFactory
+			.getLogger(LegatoServiceController.class);
+
+	private static final InstanceIdentifier<Evc> EVC_IID = InstanceIdentifier
+			.builder(MefServices.class).child(CarrierEthernet.class)
+			.child(SubscriberServices.class).child(Evc.class).build();
+
+	private static final InstanceIdentifier<SubscriberServices> EVC_IID_OPERATIONAL = InstanceIdentifier
+			.builder(MefServices.class).child(CarrierEthernet.class)
+			.child(SubscriberServices.class).build();
+
+	private ListenerRegistration<LegatoServiceController> dataTreeChangeListenerRegistration;
+
+	private static final Map<String, String> EVC_UUIDMap = new HashMap<String, String>();
+
+	private TapiConnectivityService prestoConnectivityService;
+
+	public void setPrestoConnectivityService(
+			TapiConnectivityService prestoConnectivityService) {
+		this.prestoConnectivityService = prestoConnectivityService;
+	}
+
+	public void registerListener() {
+		LOG.info("Initializing LegatoServiceController:int() ");
+
+		assert prestoConnectivityService != null;
+
+		dataTreeChangeListenerRegistration = dataBroker
+				.registerDataTreeChangeListener(new DataTreeIdentifier<Evc>(
+						LogicalDatastoreType.CONFIGURATION, EVC_IID), this);
+	}
+
+	public void close() throws Exception {
+		if (dataTreeChangeListenerRegistration != null) {
+			dataTreeChangeListenerRegistration.close();
+		}
+	}
+
+	@Override
+	public void add(DataTreeModification<Evc> newDataObject) {
+		if (newDataObject.getRootNode() != null
+				&& newDataObject.getRootPath() != null) {
+			LOG.info("  Node Added  "
+					+ newDataObject.getRootNode().getIdentifier());
+
+			Optional<Evc> OptionalEvc = LegatoUtils.readEvc(dataBroker,
+					LogicalDatastoreType.CONFIGURATION, newDataObject
+							.getRootPath().getRootIdentifier());
+
+			if (OptionalEvc.isPresent()) {
+				addNode(OptionalEvc.get());
+			}
+		}
+	}
+
+	@Override
+	public void remove(DataTreeModification<Evc> removedDataObject) {
+		if (removedDataObject.getRootNode() != null
+				&& removedDataObject.getRootPath() != null) {
+			LOG.info("  Node removed  "
+					+ removedDataObject.getRootNode().getIdentifier());
+
+			deleteNode(removedDataObject.getRootNode().getDataBefore());
+		}
+	}
+
+	@Override
+	public void update(DataTreeModification<Evc> modifiedDataObject) {
+		if (modifiedDataObject.getRootNode() != null
+				&& modifiedDataObject.getRootPath() != null) {
+			LOG.info("  Node modified  "
+					+ modifiedDataObject.getRootNode().getIdentifier());
+			Optional<Evc> OptionalEvc = LegatoUtils.readEvc(dataBroker,
```

Comment:
please use Java variable naming convention
should be ``optionalEvc`` please fix throughout the code